### PR TITLE
#176838530 bg[hotels, rooms]:  fix hotels and rooms associations 

### PR DESCRIPTION
**Pull request by SimonImanigirimpuhwe**
#### What does this PR do?

- Fix hotel model and room model associations

#### Description of Task to be completed? 

- Associate rooms model  to hotel model
- Remove unnecessary rooms endpoints
- Revisit hotel controller to include rooms when query hotel
- Revisit room controller to remove unnecessary functionality
- Change room controller functionalities
- Rewrite hotel tests
- Rewrite rooms tests
- Rewrite tests depending on hotels and rooms

#### How should this be manually tested?

- Clone the repo 

- In your home directory  cd this branch **`bg-fix-hotel-and-room-associations-#176838530`**

- Then run **`npm install`** to install all dependencies

- Run **`npm start`** to start the server

- In your default browser navigate to **`http://localhost:5000/api-docs`**   then test `rooms routes` and `hotels routes`

#### Any background context you want to provide?

- To create hotel or rooms you need first to login

#### What are the relevant Pivotal Tracker stories?

- [#176838530](https://www.pivotaltracker.com/n/projects/2476678/stories/176838530)

#### Screenshots (if appropriate)
- N/A
#### Questions:

- N/A

### CODE REVIEW

**Review by SamuelSabasNayo — APPROVED** [review]
Good work!

**Review by ConstantineMaliza — APPROVED** [review]

**Review by leny62 — APPROVED** [review]
The association is now working well.

File: src/controllers/room.js [hunk 1]
```diff
@@ -1,143 +1,99 @@
...
+    const hotel = await model.hotel.findOne({ where: { id: hotelId } });
+
+    if (!hotel) res.status(404).json({ message: res.__('Hotel with the specified ID does not exists') });
 
-    return res.status(500).send(error.message);
+    if (!room) res.status(404).json({ meassage: res.__('Room with the specified ID does not exists') });
+
+    return res.status(200).json({ room });
+  } catch (error) {
+    return res.status(500).send({ message: error.message });
```
Comment (Anguandia) [inline src/controllers/room.js]: I think at some point we will have to revise all error messages to be known beforehand and userfriendly

**Review by Anguandia — APPROVED** [review]

File: src/controllers/room.js (hunk 1 above)
Comment (SimonImanigirimpuhwe) [reply src/controllers/room.js]: Resolved

**Pull request merged**